interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/typer/MethodReferenceTyper.scala

```diff
@@ -50,46 +50,53 @@ class MethodReferenceTyper(classDefinitionSet: ClassDefinitionSet, methodExecuti
 
   private def validateClassDefinitionsNonEmpty(
       clazzDefinitions: List[ClassDefinition]
-  ): Either[Option[ExpressionParseError], List[ClassDefinition]] =
-    if (clazzDefinitions.isEmpty) Left(None) else Right(clazzDefinitions)
+  ): Either[Option[ExpressionParseError], NonEmptyList[ClassDefinition]] = {
+    NonEmptyList.fromList(clazzDefinitions).map(nel => Right(nel)).getOrElse(Left(None))
+  }
 
   private def validateMethodsNonEmpty(
-      clazzDefinitions: List[ClassDefinition]
-  )(implicit reference: MethodReference): Either[Option[ExpressionParseError], List[MethodDefinition]] = {
-    def displayableType = clazzDefinitions.map(k => k.clazzName).map(_.display).mkString(", ")
-    def isClass         = clazzDefinitions.map(k => k.clazzName).exists(_.canBeSubclassOf(Typed[Class[_]]))
+      clazzDefinitions: NonEmptyList[ClassDefinition]
+  )(implicit reference: MethodReference): Either[Option[ExpressionParseError], NonEmptyList[MethodDefinition]] = {
+    def displayableType = clazzDefinitions.map(k => k.clazzName).map(_.display).toList.mkString(", ")
+
+    def isClass = clazzDefinitions.map(k => k.clazzName).exists(_.canBeSubclassOf(Typed[Class[_]]))
 
     val clazzMethods =
-      if (reference.isStatic) clazzDefinitions.flatMap(_.staticMethods.get(reference.methodName).toList.flatten)
-      else clazzDefinitions.flatMap(_.methods.get(reference.methodName).toList.flatten)
+      if (reference.isStatic) clazzDefinitions.toList.flatMap(_.staticMethods.get(reference.methodName).toList.flatten)
+      else clazzDefinitions.toList.flatMap(_.methods.get(reference.methodName).toList.flatten)
     clazzMethods match {
       // Static method can be invoked - we cannot find them ATM
       case Nil if isClass => Left(None)
       case Nil            => Left(Some(UnknownMethodError(reference.methodName, displayableType)))
-      case methodInfos    => Right(methodInfos)
+      case first :: rest  => Right(NonEmptyList(first, rest))
     }
   }
 
   private def validateMethodParameterTypes(
-      methodInfos: List[MethodDefinition]
-  )(implicit reference: MethodReference): Either[Option[ExpressionParseError], List[TypingResult]] = {
-    // We combine MethodInfo with errors so we can use it to decide which
+      methodInfos: NonEmptyList[MethodDefinition]
```

Comment:
`methodInfos` seems like legacy from `MethodInfo` class long time ago. Maybe we could change that too? WDYT about `methodDefinitions`? Type tells us that it is `MethodDefinitions` and it is placed in private scope - so maybe just `definitions`?